src/arithmetic/bound_deducer.cc

```diff
@@ -237,9 +244,9 @@ IntSet DeduceBound(Expr v, Expr e,
   if (!d.success) return IntSet::nothing();
   Expr min = Interval::neg_inf, max = Interval::pos_inf;
   if (d.is_greater) {
-    min = d.is_equal ? d.result : d.result + 1;
+    min = d.result;
   } else {
-    max = d.is_equal ? d.result : d.result - 1;
+    max = d.result;
```

Comment:
As the deduction get complicated for checking both cases if there is equality in the comparison. I would recommend move it away by first transforming the equation to canonical form like 

```(a <= b)``` and have a single point of logic for doing the deduction

Reply:
specifically, we want to canonicalize the result if the is_equal check is false, and move to the case when is_equal is true.

Reply:
Ok, the way lt->le, gt->ge is better indeed.